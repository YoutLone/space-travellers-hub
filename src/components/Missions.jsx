import React from 'react';
import { useSelector } from 'react-redux';

function Missions() {
  const missionList = useSelector((state) => state.missions.Missions);

  return (
    <table className="container">
      <thead>
        <tr>
          <th><b>Mission</b></th>
          <th><b>Description</b></th>
          <th><b>Status</b></th>
          <th> </th>
        </tr>
      </thead>
      <tbody>
        {missionList.map((mission) => (
          <tr key={mission.mission_id} className="mission">
            <td>{mission.mission_name}</td>
            <td>{mission.description}</td>
            <td>{mission.joined}</td>
            <td> </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
}

export default Missions;
