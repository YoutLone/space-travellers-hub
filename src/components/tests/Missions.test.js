import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Missions from '../Missions';

const mockStore = configureMockStore([thunk]);

describe('Missions Component', () => {
  test('renders Missions component correctly', () => {
    const initialState = {
      missions: {
        Missions: [], // Initialize with an empty array or the default value you expect
      },
    };
    const store = mockStore(initialState);
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders Missions component with mission list correctly', () => {
    const missionList = [
      {
        mission_id: 1,
        mission_name: 'Mission 1',
        description: 'Mission 1 Description',
        joined: false,
      },
      {
        mission_id: 2,
        mission_name: 'Mission 2',
        description: 'Mission 2 Description',
        joined: true,
      },
    ];

    const initialState = {
      missions: {
        Missions: missionList, // Set the initial state with the missionList
      },
    };
    const store = mockStore(initialState);

    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
