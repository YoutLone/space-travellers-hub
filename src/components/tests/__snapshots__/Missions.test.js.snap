// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Component renders Missions component correctly 1`] = `
<table
  class="mission-container"
>
  <thead>
    <tr>
      <th>
        <b>
          Mission
        </b>
      </th>
      <th>
        <b>
          Description
        </b>
      </th>
      <th>
        <b>
          Status
        </b>
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
</table>
`;

exports[`Missions Component renders Missions component with mission list correctly 1`] = `
<table
  class="mission-container"
>
  <thead>
    <tr>
      <th>
        <b>
          Mission
        </b>
      </th>
      <th>
        <b>
          Description
        </b>
      </th>
      <th>
        <b>
          Status
        </b>
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody
    class="mission"
  >
    <tr>
      <td>
        <b>
          Mission 1
        </b>
      </td>
      <td>
        Mission 1 Description
      </td>
      <td>
        <button
          class="not-member"
          type="button"
        >
          NOT A MEMBER
        </button>
      </td>
      <td>
        <button
          class="join-mission"
          type="submit"
        >
          JOIN MISSION
        </button>
      </td>
    </tr>
  </tbody>
  <tbody
    class="mission"
  >
    <tr>
      <td>
        <b>
          Mission 2
        </b>
      </td>
      <td>
        Mission 2 Description
      </td>
      <td>
        <button
          class="active-member"
          type="button"
        >
          ACTIVE MEMBER
        </button>
      </td>
      <td>
        <button
          class="leave-member"
          type="submit"
        >
          LEAVE MISSION
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
